src/ClientCredentialsHandler.cs

```diff
@@ -1,114 +1,94 @@
 ﻿using Laserfiche.Oauth.Api.Client.Util;
-using Microsoft.Extensions.Logging;
-using Microsoft.IdentityModel.JsonWebTokens;
-using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
 using System;
+using System.IO;
 using System.Net.Http;
 using System.Net.Http.Headers;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Laserfiche.Oauth.Api.Client
 {
     public class ClientCredentialsHandler : IClientCredentialsHandler
     {
-        private HttpClient _httpClient;
-
-        private readonly ILogger _logger;
+        public HttpClient HttpClient { get; private set; }
 
         public IClientCredentialsOptions Configuration { set; get; }
 
-        public ClientCredentialsHandler(IClientCredentialsOptions configuration, IHttpClientFactory httpClientFactory = null, ILoggerFactory loggerFactory = null)
```

Comment:
Shall we keep logging?

Reply:
The current logger is just used to log errors and then we immediately throw the error. The applications that use this client library could catch the exception and log it themselves